### PR TITLE
Fix jacobian from free to curvilinear coordinates

### PR DESCRIPTION
**Pull request by FabianKlimpel**
This PR fixes the jacobian from free to curvilinear coordinates as it was missing components as was revealed by #86.

### CODE REVIEW

**Comment by asalzburger** [comment]
Clang-format failure, should be easy to fix.

**Comment by asalzburger** [comment]
I will not have time to review the maths here fully, did you validate numerically as well ?

**Comment by FabianKlimpel** [comment]
> I will not have time to review the maths here fully, did you validate numerically as well ?

The issue is that our current parametrisation will hide the issue very well. Probably it was visible in the integration tests but the changes in #86 revealed it and there it is a present component in the integration tests.
About the maths: This jacobian calculates d(Curvilinear)/d(FreeParameters). For theta we have theta = acos(Tz/sqrt(Tx^2+Ty^2+Tz^2)) with the direction vector T. It was assumed that the denominator is a constant and therefore the corresponding partial derivatives were missing.

File: Core/include/Acts/Surfaces/detail/DiscSurface.ipp [hunk 1]
```diff
@@ -77,11 +77,13 @@ inline const RotationMatrix3D DiscSurface::initJacobianToLocal(
   // Optimized trigonometry on the propagation direction
   const double x = direction(0);  // == cos(phi) * sin(theta)
   const double y = direction(1);  // == sin(phi) * sin(theta)
-  // component expressions - global
-  const double inv_sin_theta_2 = 1. / (x * x + y * y);
-  const double cos_phi_over_sin_theta = x * inv_sin_theta_2;
-  const double sin_phi_over_sin_theta = y * inv_sin_theta_2;
-  const double inv_sin_theta = sqrt(inv_sin_theta_2);
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/include/Acts/Surfaces/detail/DiscSurface.ipp:84]: Explained in the comments.

**Comment by asalzburger** [comment]
Probably better to do a rebase?

**Pull request merged**